### ISSUE TITLE
support for building packages that depend on other aur packages

### ISSUE THREAD

**Issue by mariuszste** [VirtCode/serene-aur#4]
let's say I wanted to build `adw-gtk-theme` but it depends on `adw-gtk3` that is only available in the AUR. I have `adw-gtk3` added as a separate package and it built successfully but even after that `adw-gtk3` still fails to build.

**Comment by VirtCode** [comment]
Yes, serene does not yet properly support dependencies, and does not resolve them itself.

Adding dependencies manually is however already sort-of supported. You can use set `REPOSITORY_URL` environment variable (see in the [docs](https://github.com/VirtCode/serene-aur/blob/main/server/README.md#configuration)) to an URL to the server for build containers to access it for dependencies. The server will then add that URL as a Pacman repository to the build containers, so make sure you provide a URL which is valid for the containers, the easiest option would just be the normal URL to the server. The server can't determine the URL automatically, as it strongly depends on the actual deployment which is always a bit different.

I know that this solution is not ideal, and automatically adding dependencies from the AUR should definitely be supported. I'm planning to implement this some time in the future. For now, you'll have to add the dependencies manually, wait for them to build, then rebuild the dependents.

**Comment by dumbasPL** [comment]
> You can use set REPOSITORY_URL environment variable

didn't notice that, thanks

Another thing worth mentioning is that packages that don't specify the version of the dependency expect the latest version of that dependency. This is especially true for `-git` packages using other `-git` packages. For fast-moving projects building one without updating the other often results in problems or even failed builds. So keeping the right update order would definitely be a thing to consider

**Comment by VirtCode** [comment]
I think #15 would be finally ready by now (still testing a bit myself). If you want you can try it out, or leave other feedback if you have any. Note that the migration can be a bit flakey if you have a ton of packages (see my comment in the PR), but should otherwise work. I'm intending to merge it in a couple of days.
